<launch>
  <arg name="port" default="/dev/rfcomm0" />
  <arg name="baud" default="57600" />
  <arg name="component_name" default="hand" />
  
  <group ns="$(arg component_name)">
    <remap from="internal/recover" to="driver/recover"/>
    <remap from="internal/halt" to="driver/halt"/>
    
    <include file="$(find cob_hand_bridge)/launch/serial_bridge.launch" ns="internal">
        <arg name="port" value="$(arg port)"/>
        <arg name="baud" value="$(arg baud)"/>
    </include>

    <node pkg="cob_hand_bridge" type="cob_hand_bridge_node" name="cob_hand_bridge">
        <rosparam param="sdhx/joint_names" subst_value="True">[$(arg component_name)_finger_1_joint, $(arg component_name)_finger_2_joint]</rosparam>
    </node>
  </group>
</launch>
